### PR TITLE
Adjust string and bytes records' field names

### PR DESCRIPTION
**Pull request by priyank23**
This refers to issue #15219.
This simplifies the names of fields in the `string` and `bytes` record of chapel. Following are the names of the fields that are modified -
* len -> buffLen
* _size -> buffSize
* isowned -> isOwned
* buf -> buff

This makes the field names much more clearer to comprehend and more consistent to the chapel's coding standards.

### CODE REVIEW

**Comment by priyank23** [comment]
@e-kayrakli Apologies from my side. There was an issue I couldn't resolve in the previous PR #15225. I have implemented all the fixes in this PR. Also, I have ran the tests stated by you and resolved the conflicts. Kindly, review.

File: modules/internal/Bytes.chpl [hunk 1]
```diff
@@ -381,68 +381,68 @@ module Bytes {
...
-          if this.isowned && !this.isEmpty() then
+          if this.isOwned && !this.isEmpty() then
             bufferFree(this.buff);
-          this.buff = buf;
-          this._size = size;
+          this.buff = buff;
+          this.buffSize = size;
         }
       } else {
         // If s_len is 0, 'buf' may still have been allocated. Regardless, we
```
Comment (e-kayrakli) [inline modules/internal/Bytes.chpl:415]: buf->buff

File: modules/internal/BytesStringCommon.chpl [hunk 2]
```diff
@@ -178,81 +178,81 @@ module BytesStringCommon {
...
       if otherRemote {
         // if s is remote, copy and own the buffer
         x.buff = bufferCopyRemote(other.locale_id, other.buff, otherLen);
-        x._size = otherLen+1;
+        x.buffSize = otherLen+1;
       }
       else {
         // if s is local create a copy of its buffer and own it
         const (buf, allocSize) = bufferCopyLocal(other.buff, otherLen);
         x.buff = buf;
```
Comment (e-kayrakli) [inline modules/internal/BytesStringCommon.chpl:253]: Small nit, but let's use `buff` for the temporary variable, as well

File: modules/internal/BytesStringCommon.chpl [hunk 3]
```diff
@@ -466,31 +466,31 @@ module BytesStringCommon {
...
       if joinedSize == 0 then
         return '';
 
       var joined: t;
-      joined.len = joinedSize;
+      joined.buffLen = joinedSize;
 
-      var (newBuf, allocSize) = bufferAlloc(joined.len+1);
-      joined._size = allocSize;
+      var (newBuf, allocSize) = bufferAlloc(joined.buffLen+1);
```
Comment (e-kayrakli) [inline modules/internal/BytesStringCommon.chpl:478]: newBuff

**Comment by e-kayrakli** [comment]
I'll start testing this tomorrow and let you know if there are any other test that fails. Otherwise, I don't see anything wrong with the code changes at the moment.

Meanwhile, could you update the PR message (not the title) to give a short-and-sweet summary of what this PR is about besides linking to the issue? The gist of is in my own view is that it makes string and bytes records' field names clearer and more consistent with our coding standards. You should probably also add a bullet list for the fields whose names that are changed from X to Y.

That PR message will be the merge commit message and will be in the git logs forever, so it is kinda important for us ;)

Thanks!

**Comment by e-kayrakli** [comment]
I did a `CHPL_COMM=gasnet` paratest and the following tests have failed:

```
[Error matching compiler output for interop/C/multilocale/exportString/noArgsRetString]
[Error matching compiler output for interop/C/multilocale/exportString/stringArgsRetString]
[Error matching compiler output for interop/C/multilocale/exportString/stringArgsRetVoid]
[Error matching compiler output for io/cassella/readline]
[Error matching compiler output for library/packages/ZMQ/interop-py/server]
[Error matching compiler output for library/packages/ZMQ/multilocale/hello-multilocale]
[Error matching compiler output for library/packages/ZMQ/multilocale/req-rep]
[Error matching compiler output for multilocale/strings/dmk/utf8_misc]
[Error matching compiler output for multilocale/strings/dmk/utf8_values]
[Error matching compiler output for patterns/dynamic-programming/sequential/edit_distance]
[Error matching compiler output for patterns/dynamic-programming/sequential/longest_common_subsequence]
[Error matching compiler output for release/examples/benchmarks/shootout/revcomp-fast]
[Error matching compiler output for release/examples/benchmarks/shootout/revcomp]
[Error matching compiler output for studies/shootout/k-nucleotide/bradc/knucleotide-blc]
[Error matching compiler output for studies/shootout/reverse-complement/bharshbarg/revcomp-begin]
[Error matching compiler output for studies/shootout/reverse-complement/bharshbarg/revcomp-buf]
[Error matching compiler output for studies/shootout/reverse-complement/bharshbarg/revcomp-line]
[Error matching compiler output for studies/shootout/reverse-complement/bharshbarg/revcomp-mark-skip-read-begin]
[Error matching compiler output for trivial/deitz/demo/sample2_byteIndex]
[Error matching compiler output for types/string/StringImpl/memLeaks/substring_byte]
[Error matching compiler output for types/string/bwross/string_byte_index]
[Error matching compiler output for types/string/deitz/test_string_byte_select1]
[Error matching compiler output for types/string/dmk/utf8_misc]
[Error matching compiler output for types/string/dmk/utf8_values]
[Error compilation failed for types/string/psahabu/perf/allocate]
[Error compilation failed for types/string/psahabu/perf/arguments]
[Error matching compiler output for types/string/validation/nonUTFEscape]
```

I did a quick check and none seemed surprising to me. Just more of the same changes you have already made. I hope that the number of places you gonna have to touch is less than the number of failures.

I think this also is a good list of tests that we should look into because they may point to some standard/package modules fiddling with `string` internals that might be avoided. Let me know if any of the failures are not clear.

**Comment by priyank23** [comment]
> [Error matching compiler output for interop/C/multilocale/exportString/noArgsRetString
[Error matching compiler output for interop/C/multilocale/exportString/stringArgsRetString]
[Error matching compiler output for interop/C/multilocale/exportString/stringArgsRetVoid]

@e-kayrakli How to run test for these files as this is not `.chpl` file. 

> [Error matching compiler output for library/packages/ZMQ/interop-py/server]
[Error matching compiler output for library/packages/ZMQ/multilocale/hello-multilocale]
[Error matching compiler output for library/packages/ZMQ/multilocale/req-rep]

These are showing the following error.
 `modules/packages/ZMQHelper/zmq_helper.h:24:10: fatal error: zmq.h: No such file or directory`
How to resolve this?

I have passed all the other tests successfully.

**Comment by e-kayrakli** [comment]
For the tests that don't have .chpl, they are C-based tests that tests our C interop layer. You should run the `.test.c` file instead. The errors that you see from the start_test might be confusing, but they must be using a `.chpl` file as a library (I think I know those tests and they use `TestLibrary.chpl` in the same dir or something) which may have the errors that you need to fix.

For the ZMQ errors, if you are interested in getting success on those tests, you need to install ZeroMQ on your system: https://chapel-lang.org/docs/master/modules/packages/ZMQ.html. As far as I recall it was pretty easy, but if I am remembering it wrong let me know instead of spending hours on it. We'll figure out something else

**Comment by priyank23** [comment]
@e-kayrakli There are no ZMQ errors now. In case of C-interop tests, upon running `start_test test/interop/C/multilocale/exportString/stringArgsRetVoid.ml-test`, I am getting `[Skipping multilocale-only c test: interop/C/multilocale/exportString/stringArgsRetVoid]`. What should I do?

**Comment by e-kayrakli** [comment]
You need to configure your Chapel with multilocale setup and rebuild. See https://chapel-lang.org/docs/master/platforms/udp.html and let me know if you have any issues.

**Comment by priyank23** [comment]
@e-kayrakli Thanks. All the tests are now passing successfully.

**Comment by e-kayrakli** [comment]
Great! To be sure, are all the changes pushed to this branch? I am asking because your last commit that I can see here (595be54) predates our discussions

I'll run local and gasnet tests again

**Comment by priyank23** [comment]
Yes! I was unable to check before due to system configurations. After configuring as per your guidelines, all the tests passed without any changes to the previous commit.

**Comment by e-kayrakli** [comment]
Nice. Firing tests:

- [ ] standard
- [ ] gasnet

**Comment by e-kayrakli** [comment]
Testing is going fine but you'll need to fix the conflict after https://github.com/chapel-lang/chapel/pull/15525 went in yesterday. It should be very easy to fix, but will require you to merge master on this branch. Ideally you can rebase master instead but it may be more complicated. It is generally a straightforward process but may get confusing if it is the first time you are doing this. Let me know if you need assistance.

p.s. apparently github has a tool that allows you to resolve conflicts through the web interface, but never used it myself nor heard any experiences about it.

**Comment by priyank23** [comment]
@e-kayrakli Fixed them. Thanks for the update.

**Comment by e-kayrakli** [comment]
I did one more standard paratest just to be sure. Also, edited the PR title to save one more back-and-forth, hope you don't mind. Merging now.

**Pull request merged**